features/step_definitions/user_steps.rb

```diff
@@ -42,39 +42,44 @@
 Given(/^I am logged into the site$/) do
   visit new_user_session_path
   FactoryGirl.create(:user)
-  fill_in 'Email', with: 'newuser@sb.com' 
-  fill_in 'Password', with: 'password' 
+  fill_in 'Email', with: 'newuser@sb.com'
+  fill_in 'Password', with: 'password'
   click_on 'Log in'
 end
 
 When(/^I visit my own profile page$/) do
-  pending # express the regexp above with the code you wish you had
+ visit root_path
 end
 
 Then(/^I see my username$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content("NewUser")
 end
 
 Then(/^I can see my avatar$/) do
-  pending # express the regexp above with the code you wish you had
+  visit page.find('#avatar')[:src]
+  page.status_code.should be 200
 end
 
 Then(/^I can see my bio$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content("This is my bio.")
 end
 
 Then(/^I can see the number of tweets I have made$/) do
-  pending # express the regexp above with the code you wish you had
+  click_on 'Create Tweet'
+  fill_in 'Text', with: 'This is a Tweet'
+  click_on 'Create Tweet'
+  visit root_path
+  expect(page).to have_content("1 Tweets")
 end
```

Comment:
In `Then` blocks, you shouldn't actually be doing any page interactions. The state of the page should be set up by previous `When`s. All you should be doing here is the expect line.